addons/hr_contract/models/hr_employee.py

```diff
@@ -138,6 +139,29 @@ def _get_calendars(self, date_from=None):
                 res[employee.id] = contracts[0].resource_calendar_id.sudo(False)
         return res
 
+    def _get_calendar_periods(self, start, stop):
```

Comment:
i would still call super here because if we don't its a change of behavior, overtime is computed even if you don't have a contract, so unavailability should work this way as well @mepe-odoo

Reply:
@sofiagvaladze I don't agree, because if you don't have a contract you are not supposed to work. it was the same in working schedule task :)

Reply:
@jugj-odoo but If you work without contract, it will be an overtime no ? :thinking:

Reply:
no it will be computed as normal hours based on your calendar + overtime if you do more, attendance overtime does not take contracts into account

Reply:
ok nvm we'll figure this later, leave it like that for now